core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -350,6 +353,11 @@ private String getNextTasksToBeScheduled(Workflow workflow, Task task) {
     Task retry(TaskDef taskDefinition, WorkflowTask workflowTask, Task task, Workflow workflow) throws TerminateWorkflowException {
 
         int retryCount = task.getRetryCount();
+
+        if(taskDefinition == null) {
+            taskDefinition = metadataDAO.getTaskDef(task.getTaskType());
```

Comment:
`metadataDAO.getTaskDef` can return a null value and this can still fail.
We can possibly throw an exception here,
```java
taskDefinition = Optional.ofNullable(taskDefinition)
                .orElseGet(() -> Optional.ofNullable(metadataDAO.getTaskDef(task.getTaskType()))
                        .orElseThrow(() -> new TerminateWorkflowException(String.format("Invalid task specified. Cannot find task by name %s in the task definitions", task.getTaskDefName())));
```